dockershim: support annotations and filtering by sandbox ID

yujuhong opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32835)

<!-- Reviewable:end -->

yujuhong committed:
dockershim: fix sandbox/container filtering

yujuhong committed:
dockershim: add support for annotations

yujuhong committed:
dockershim: support filter containers by sandbox ID

feiskyer commented on pkg/kubelet/dockershim/helpers.go:
```diff
@@ -30,6 +30,10 @@ import (
 )
 
+const (
+	annotationPrefix = "annotation."
```
do we need to make `annotationPrefix` longer, e.g. `io.kubernetes.annotation.`?

yujuhong replied:
We can do that, but I was afraid that this would grow too long. E.g., the annotation key could be `io.kubernetes.terminationmessage`. After this, it becomes `io.kubernetes.annotation.io.kubernetes.terminationmessage`

yujuhong committed:
CRI: add annotations to PodSandbox

Both Container and ContainerStatus includes labels and annotations. PodSandbox
should be consistent with that.

yujuhong commented:
Add a new commit to add "Annotations" to "PodSandbox" as well.

Random-Liu commented on pkg/kubelet/dockershim/docker_sandbox_test.go:
```diff
@@ -46,8 +52,11 @@ func TestListSandboxes(t *testing.T) {
+		c := makeSandboxConfigWithLabelsAndAnnotations(fmt.Sprintf("%s%d", name, i),
+			fmt.Sprintf("%s%d", namespace, i), fmt.Sprintf("%d", i), 0,
+			map[string]string{"label": fmt.Sprintf("sandbox%d", i)},
+			map[string]string{"annotation": fmt.Sprintf("sandbox%d", i)},
```
nit: would be better to make test label and annotation different.

yujuhong replied:
fixed.

Random-Liu commented:
LGTM with one nit.
